wherobots/db/connection.py

```diff
@@ -155,12 +155,20 @@ def __listen(self):
 
             query.state = ExecutionState.COMPLETED
             if result_format == ResultsFormat.JSON:
-                query.handler(json.loads(result_bytes.decode("utf-8")))
+                data = json.loads(result_bytes.decode("utf-8"))
+                columns = data["columns"]
+                column_types = data.get("column_types")
```

Comment:
Is `column_types` optional? If so, then it's good that you're using `data.get()` here, but then in `Cursor.__get_results` you expect `column_types` to be non-None. You either need to ensure `column_types` is always provided, or change `__get_results` to be more defensive.